### PR TITLE
Use ? instead of unwrap()

### PR DESCRIPTION
**Pull request by kzys**
This change addresses #833, while there are still some unwrap() though.

### CODE REVIEW

**Comment by kzys** [comment]
Regarding https://solana-ci-gate.herokuapp.com/buildkite_public_log?https://buildkite.com/solana-labs/solana/builds/2088, I'd like to know rustc's versions on the CI hosts.

**Comment by mvines** [comment]
@kzys thanks, yeah it's not this PR.   Sorry for the speed bump.  #871 covers updating those machines

File: tests/data_replicator.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ where
 }
 /// ring a -> b -> c -> d -> e -> a
 #[test]
-fn gossip_ring() {
+fn gossip_ring() -> Result<(), solana::result::Error> {
```
Comment (rob-solana) [inline tests/data_replicator.rs]: where do we define solana::result::Error?  just curious, might be useful for #822

Comment (garious) [reply tests/data_replicator.rs]: You can tighten up that code with:

```rust
use solana::result;
fn gossip_ring() -> result::Result {
```

**Comment by garious** [comment]
@aeyakovenko, what do you think about this? My feeling is that when you're using either `unwrap()` or `?`, it's part of the test that you're not intending to test, and should almost always succeed. When it doesn't succeed, debugging it with `RUST_BACKTRACE=1` is the normal dev process, and where you'd get that line number. Therefore, I'm leaning toward preferring the use of `?` over `unwrap()`, as it makes the unit tests more readable. For me, readability is a higher priority than non-RUST_BACKTRACE debuggability. Thoughts?

**Comment by garious** [comment]
@kzys, looks like we'll need to you rebase on the latest `multinode.rs`. Thanks for taking this on!

**Comment by aeyakovenko** [comment]
how do you see RUST_BACKTRACE=1 helping?   `?` wont panic and you will go up the stack out of the test.

**Comment by kzys** [comment]
You can see the actual backtrace on https://github.com/solana-labs/solana/issues/833

I can't recall that the line number on the stacktrace points ? or the function itself. I can take a look that tonight.

File: tests/data_replicator.rs [hunk 2]
```diff
@@ -164,7 +166,7 @@ pub fn crdt_retransmit() {
     assert!(done);
     let mut b = Blob::default();
     b.meta.size = 10;
-    Crdt::retransmit(&c1, &Arc::new(RwLock::new(b)), &tn1).unwrap();
+    Crdt::retransmit(&c1, &Arc::new(RwLock::new(b)), &tn1)?;
```
Comment (aeyakovenko) [inline tests/data_replicator.rs]: if this function fails, you wont see it in the stackframe.  unless the Err itself is generated with a line number.

**Comment by kzys** [comment]
I've updated the pull request to rebase and use `solana::result::Result<>`

**Comment by garious** [comment]
@mvines, what's up with Grimes?

**Comment by mvines** [comment]
> @mvines, what's up with Grimes?

what do you mean?

**Comment by garious** [comment]
@mvines, I mean that each time we tried to add the CI label, Grimes removed it.

**Comment by mvines** [comment]
Ah so that's by design.  The `ci` label is only good for the particular commit it was added to, so it's removed immediately after ci is started for that commit.   We can certainly discuss if that's the right design or not on discord though!

**Pull request merged**